xl2times/transforms.py

```diff
@@ -772,39 +787,52 @@ def process_time_periods_table(table: datatypes.EmbeddedXlTable):
     return [process_time_periods_table(table) for table in tables]
 
 
-def generate_all_regions(
+def process_regions(
     config: datatypes.Config,
     tables: List[datatypes.EmbeddedXlTable],
+    model: datatypes.TimesModel,
 ) -> List[datatypes.EmbeddedXlTable]:
     """
     Include IMPEXP and MINRNW together with the user-defined regions in the AllRegions set
     IMPEXP and MINRNW are external regions that are defined by default by Veda
     """
 
-    external_regions = ["IMPEXP", "MINRNW"]
-    df = pd.DataFrame(external_regions, columns=["region"])
+    model.all_regions.update((["IMPEXP", "MINRNW"]))
+    model.internal_regions.update(
+        utils.single_column(tables, datatypes.Tag.book_regions_map, "region")
+    )
+    model.all_regions.update(model.internal_regions)
 
-    for table in tables:
-        if table.tag == datatypes.Tag.book_regions_map:
-            df = pd.concat([df, table.dataframe])
+    # Apply regions filter
+    if config.filter_regions:
+        keep_regions = model.internal_regions.intersection(config.filter_regions)
+        if keep_regions:
+            model.internal_regions.intersection_update(keep_regions)
```

Comment:
```suggestion
            model.internal_regions = keep_regions
```
This should have the same behaviour, right? (And avoid calculating intersection again.)

Reply:
I hope so. :-) My thinking here was to stick to using class methods, just in case. But if we can avoid doing repeated calculations...